expression-language.md

```diff
@@ -0,0 +1,375 @@
+| `x = y: String x String -> Boolean`     | Returns `true` if the values of `x` and `y` are equal                                           |
+| `x != y: String x String -> Boolean`    | Same as `NOT (x = y)`                                                                           |
+| `x <> y: String x String -> Boolean`    | Same as `NOT (x = y)`                                                                           |
+
+#### 3.4.3. Like operator
+
+| Definition                                       | Semantics                                           |
+| ------------------------------------------------ | --------------------------------------------------- |
+| `x LIKE pattern: String x String -> Boolean`     | Returns `true` if the value x matches the `pattern` |
+| `x NOT LIKE pattern: String x String -> Boolean` | Same as `NOT (x LIKE PATTERN)`                      |
+
+The pattern of the `LIKE` operator can contain:
+
+- `%` represents zero, one, or multiple characters
+- `_` represents a single character
+
+For example, the pattern `_b*` will accept values `ab`, `abc`, `abcd1` but won't accept values `b` or `acd`.
+
+Both `%` and `_` can be escaped with `\`, in order to be matched literally. For example, the pattern `abc\%` will match
+`abc%` but won't match `abcd`.
+
+#### 3.4.4. Exists operator
+
+| Definition                          | Semantics                                                                   |
+| ----------------------------------- | --------------------------------------------------------------------------- |
+| `EXISTS identifier: Any -> Boolean` | Returns `true` if the attribute `identifier` exists in the input CloudEvent |
+
+Note: `EXISTS` MUST always return `true` for the required context attributes because the input CloudEvent is always
+assumed valid, e.g. `EXISTS id` MUST always return `true`.
+
+#### 3.4.5. In operator
+
+| Definition                                             | Semantics                                                                |
+| ------------------------------------------------------ | ------------------------------------------------------------------------ |
+| `x IN (y1, y2, ...): String x String^n -> Boolean`     | Returns `true` if `x` is an element included in the set of `yN` elements |
+| `x NOT IN (y1, y2, ...): String x String^n -> Boolean` | Same as `NOT (x IN set)`                                                 |
+
+The set of values `yN` can be of arbitrary length, and the matching is done using the same semantics of the equal `=`
+operator.
+
+### 3.5. Functions
+
+The following tables show the built-in functions that MUST be supported by a CESQL evaluator.
+
+#### 3.5.1. Casting and type checking
+
+| Definition                      | Semantics                                                                                                                        |
```

Comment:
You are punting on type inference here. I realize that's a grueling topic, but type casts should be a last resort and we should avoid having them completely, IMO.

Reply:
I agree, but I think this is unavoidable because CloudEvent extensions don't have type annotations. They can only be inferred from the lexical value, which sometimes leads to ambiguous case, eg. `true as string` vs `true as boolean`.

Reply:
For me explicit type casting should definitely be there, for the reasons mentioned by @lionelvillard. Given that, as discussed during the meeting, I'll try to draft some inference rules here